### PR TITLE
add profiler to php-dev

### PR DESCRIPTION
**Pull request by ChrisB9**
related: https://github.com/pluswerk/docker-global/pull/21
closes: #58

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
   echo "%sudo ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
   curl -fsSL https://get.docker.com/ | sh && \
   rm -rf /var/lib/apt/lists/*
 
+# Install XHProf
+RUN cd /tmp && wget https://github.com/tideways/php-xhprof-extension/archive/v${XPROF_VERSION}.zip && unzip v${XPROF_VERSION}.zip && ls && \
+    cd php-xhprof-extension-${XPROF_VERSION} && phpize && ./configure && make && make install
+
 USER application
 RUN composer global require hirak/prestissimo davidrjonas/composer-lock-diff
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
RUN composer global require hirak/prestissimo davidrjonas/composer-lock-diff perftools/xhgui-collector && composer clear
```

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+RUN composer global require perftools/xhgui-collector
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
   apt-get install -y sudo vim nano less tree bash-completion mysql-client iputils-ping && \
   usermod -aG sudo application && \
   echo "%sudo ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
   curl -fsSL https://get.docker.com/ | sh && \
   rm -rf /var/lib/apt/lists/*
 
+# Install XHProf
+RUN cd /tmp && wget https://github.com/tideways/php-xhprof-extension/archive/v${XPROF_VERSION}.zip && unzip v${XPROF_VERSION}.zip && ls && \
+    cd php-xhprof-extension-${XPROF_VERSION} && phpize && ./configure && make && make install
+
```
Comment (Kanti) [inline Dockerfile]:
```suggestion

RUN echo "extension=tideways_xhprof.so" >> /opt/docker/etc/php/php.ini
COPY profiler.php /opt/docker/profiler.php
RUN echo "auto_prepend_file = /opt/docker/profiler.php" >> /opt/docker/etc/php/php.ini

```

File: Dockerfile [hunk 4]
```diff
@@ -25,6 +33,10 @@ RUN echo "source ~/.additional_bashrc.sh" >> ~/.bashrc
 
 USER root
 
+RUN echo "extension=tideways_xhprof.so" >> /opt/docker/etc/php/php.ini
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 5]
```diff
@@ -25,6 +33,10 @@ RUN echo "source ~/.additional_bashrc.sh" >> ~/.bashrc
...
+COPY profiler.php /opt/docker/profiler.php
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 6]
```diff
@@ -25,6 +33,10 @@ RUN echo "source ~/.additional_bashrc.sh" >> ~/.bashrc
...
+RUN echo "auto_prepend_file = /opt/docker/profiler.php" >> /opt/docker/etc/php/php.ini
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 7]
```diff
@@ -25,6 +33,10 @@ RUN echo "source ~/.additional_bashrc.sh" >> ~/.bashrc
...
+
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
```

File: run_tests.sh [hunk 8]
```diff
@@ -2,6 +2,9 @@
 
 # php should be callable
 php -v &&
+# php should have speficic modules installed
+php -m | grep xdebug || exit 1 &&
+php -m | grep tideways || exit 1 &&
```
Comment (Kanti) [inline run_tests.sh]: only if php >=7.0

**Comment by Kanti** [comment]
@Kanti Better Docs needed.

File: Dockerfile [hunk 9]
```diff
@@ -33,7 +41,9 @@ RUN curl https://raw.githubusercontent.com/git/git/v$(git --version | awk 'NF>1{
 # add .additional_bashrc.sh
 COPY bin/* /usr/local/bin/
 COPY .additional_bashrc.sh /home/application/.additional_bashrc.sh
+COPY .additional_bashrc.sh /home/root/.additional_bashrc.sh
```
Comment (ChrisB9) [inline Dockerfile]: In additional bash RC are sudos, that will not work with root

File: run_tests.sh [hunk 10]
```diff
@@ -2,6 +2,9 @@
 
 # php should be callable
 php -v &&
+# php should have speficic modules installed
+php -m | grep xdebug || exit 1 &&
+[ 70000 -le $(php -r "echo PHP_VERSION_ID;") ] || php -m | grep tideways || exit 1 &&
```
Comment (ChrisB9) [inline run_tests.sh]:
```suggestion
[ 70000 -le $(php -r "echo PHP_VERSION_ID;") ] || php -m | grep tideways || exit 1 &&
php -m | grep mongo || exit 1 &&
```

Comment (Kanti) [reply run_tests.sh]: mongo is not installed anymore as it is not needed.

**Pull request merged**